Automattic/jetpack
# Replace mysql entrypoint to do log dir chmod and auto-update

This is an update to #20226 to make it work for me.

#### Changes proposed in this Pull Request:

Instead of spawning a separate container, we can run our own script that
does a little work before execing the normal entrypoint.

* Fix ownership of the logs directory.
* In a background subshell, run the updater once the DB comes up.

Note a drawback to this approach is that we may need to update our custom script if
the upstream MariaDB image moves their entrypoint or changes their default CMD.

#### Jetpack product discussion


See #20226

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Shut down any existing Jetpack Docker env, e.g. with `jetpack docker down`
* Create a fresh clone of the repo, checked out to master.
* `jetpack docker up`, then close it down.
* Check out this PR.
* Do `jetpack docker up`. Check for the following:
  * See that the logs for db_1 indicate that an upgrade was run.
  * See that there are no complaints about `/var/log/mysql/slow.log` being unable to be accessed.
  * On the host, see that `tools/docker/logs/jetpack_dev/mysql/slow.log` exists.
* Close down the env, then `jetpack docker up` again.
  * See that the logs for db_1 indicate that an upgrade was attempted.
  * See that there are still no complaints about `/var/log/mysql/slow.log` being unable to be accessed.
